### PR TITLE
t2 provision refusal should occur before any connection attempt is made. Fixes gh-168

### PR DESCRIPTION
**Pull request by rwaldron**
See #168 for explanation.

This patch is broken into several commits to make the process of solving the problem a little clearer. Once approved, I will squash.

### CODE REVIEW

File: lib/tessel/provision.js [hunk 1]
```diff
@@ -9,7 +10,22 @@ var commands = require('./commands');
 var logs = require('../logs');
 var dir = osenv.home() + '/.tessel';
 var filename = 'id_rsa';
-var filepath = dir + '/' + filename;
+var filepath = path.join(dir, filename);
```
Comment (johnnyman727) [inline lib/tessel/provision.js]: Ah cool! I didn't know about `path.join`.

File: lib/tessel/provision.js [hunk 2]
```diff
@@ -102,6 +111,7 @@ function authTessel(tessel) {
 
   // Make sure Tessel has the authFile
   checkAuthFileExists(tessel, authFile, function() {
+    console.log("message");
```
Comment (johnnyman727) [inline lib/tessel/provision.js]: Runaway log.

**Comment by johnnyman727** [comment]
This is going to be real fun to rebase with #167. :/

File: bin/tessel-2.js [hunk 3]
```diff
@@ -22,6 +22,11 @@ parser.command('provision')
         process.exit(1);
       });
   })
+  .option('force', {
```
Comment (johnnyman727) [inline bin/tessel-2.js]: I updated this feature's spec issue: https://github.com/tessel/t2-cli/issues/64

**Comment by johnnyman727** [comment]
Looks good. Once you:
-[ ] Remove legacy `console.log`
-[ ] Fix styling errors that is failing Travis

we can merge. I'll merge #167 afterward.

File: bin/tessel-2.js (hunk 3 above)
Comment (rwaldron) [inline bin/tessel-2.js]: Thanks!

**Comment by johnnyman727** [comment]
Awesome, thanks @rwaldron!

File: test/unit/provision.js [hunk 4]
```diff
@@ -0,0 +1,135 @@
...
+};
+
+exports['controller.provisionTessel'] = {
+  setUp: function(done) {
+    this.isProvisioned = sinon.stub(Tessel, 'isProvisioned').returns(true);
+    this.provisionTessel = sinon.spy(controller, 'provisionTessel');
+    this.exec = sinon.stub(cp, 'exec', function(command, callback) {
+      callback();
+    });
+    var provisionSpy = sinon.spy();
```
Comment (johnnyman727) [inline test/unit/provision.js]: @rwaldron Should this be spying on something in particular?

**Pull request merged**